theseus/embodied/robot/link.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Optional, Any
+import torch
+
+
+class Link(abc.ABC):
+    def __init__(
+        self,
+        name: str,
+        id: int = -1,
+        parent: Any = None,
+        children: Optional[List[Any]] = None,
```

Comment:
Is it `Any` or `Union[Link, Joint]`?

Reply:
It should `Joint`. But due to circular dependence, it seems impossible to `import Joint` in `Link`.

Reply:
Moved `Link` to `joint.py` and now `Any->Joint`